### PR TITLE
support legacy clusterset 

### PR DESCRIPTION
**Pull request by haoqing0110**
Ref: https://github.com/stolostron/backlog/issues/15278

This PR integrates placement with new clusterset api https://github.com/open-cluster-management-io/api/pull/141. 

`ManagedClusterSets` with incorrect `SelectorType` (currently the type is `LegacyClusterSetLabel` only) will not be returned by func `getEligibleClusterSets()`.

Signed-off-by: haoqing0110 <qhao@redhat.com>

### CODE REVIEW

File: pkg/controllers/scheduling/scheduling_controller.go [hunk 1]
```diff
@@ -333,6 +333,22 @@ func (c *schedulingController) getEligibleClusterSets(placement *clusterapiv1bet
...
+			nonLegacyClusterSetNames.Insert(clusterSetName)
+			continue
+		}
+
+		selectorType := clusterSet.Spec.ClusterSelector.SelectorType
+		if len(selectorType) > 0 && selectorType != clusterapiv1beta1.LegacyClusterSetLabel {
+			nonLegacyClusterSetNames.Insert(clusterSetName)
+		}
+	}
+	clusterSetNames = clusterSetNames.Difference(nonLegacyClusterSetNames)
```
Comment (qiujian16) [inline pkg/controllers/scheduling/scheduling_controller.go:350]: I do not think we should ignore clusterset here. I'd rather we have a logic in getAvailableClusters that only select clusters from legacy cluster set

Comment (haoqing0110) [reply pkg/controllers/scheduling/scheduling_controller.go]: moved the logic to getAvailableClusters

File: pkg/controllers/scheduling/scheduling_controller.go [hunk 2]
```diff
@@ -359,8 +343,25 @@ func (c *schedulingController) getAvailableClusters(clusterSetNames []string) ([
 	if len(clusterSetNames) == 0 {
 		return nil, nil
 	}
+
+	// filter avilable clustersets
```
Comment (qiujian16) [inline pkg/controllers/scheduling/scheduling_controller.go:347]: I think you would want a 
```go
avilableClusterSetsMap := map[string]sets.String

for _, name := range clusterSetNames {
		// ignore clusterset if failed to get
		clusterSet, err := c.clusterSetLister.Get(name)
		if err != nil {
			continue
		}

		// clusterset ClusterSelector type is empty or LegacyClusterSetLabel is treated as avaliable
		selectorType := clusterSet.Spec.ClusterSelector.SelectorType
		if len(selectorType) == 0 || selectorType == clusterapiv1beta1.LegacyClusterSetLabel {
			if avilableClusterSetsMap[clusterSetLabel] == nil {
                             avilableClusterSetsMap[clusterSetLabel]  = sets.NewString()
                         }
                        avilableClusterSets.Insert(name)
		}
	}

// now list all clusters
for labels, avilableClusterSets := range avilableClusterSetsMap {
  requirement, err := labels.NewRequirement(clusterSetLabel, selection.In, avilableClusterSets.List())
  if err != nil {
    // log what is the error
		continue
  }

  labelSelector := labels.NewSelector().Add(*requirement)

  clusters, err := c.clusterLister.List(labelSelector)

  //join all results here.
}

```

**Pull request merged**